README.md

```diff
@@ -1,210 +1,261 @@
+`mvn test` runs all tests with JDK8, JDK11, and then with JDK17
+To run the tests with only one particular JDK version run:
+* for JDK8:
+```
+mvn surefire:test@test-default
+```
+or
+```
+mvn surefire:test
+```
+* for JDK11:
+```
+mvn surefire:test@test-jdk11
+```
+* for JDK17:
+```
+mvn surefire:test@test-jdk17
+```
+
+# Use
+
+
+The best way to use `MemoryMeter` is to start the JVM with "-javaagent:<path to>/jamm.jar" in order to use the
+`Instrumentation` strategy to guess objects' sizes.
+
+`MemoryMeter` can be used in your code like this:
 
     MemoryMeter meter = MemoryMeter.builder().build();
     meter.measure(object);
     meter.measureDeep(object);
 
-See [`MemoryMeter.Builder`](./src/org/github/jamm/MemoryMeter.java) for more
-options.
 
 If you would like to use `MemoryMeter` in a web application, make sure
 that you do NOT put this jar in `WEB-INF/lib`, as that may cause problems
-since your code is accessing a MemoryMeter from a different class loader
+since your code is accessing a `MemoryMeter` from a different class loader
 than the one loaded by the `-javaagent` and won't see it as initialized.
 
 If you want `MemoryMeter` not to measure some specific classes, you can
-mark the classes (or interfaces) using the
+mark the classes (or interfaces) using the `Unmetered` annotation.
+
+# The Maven coordinates for the latest version of Jamm
+
+```
+  <groupId>com.github.jbellis</groupId>
+  <artifactId>jamm</artifactId>
+  <version>0.4.0-SNAPSHOT</version>
+```
+
+# 0.4.0 breaking changes
+
+The 0.4.0 version comes with speed improvements and support java versions up to Java 17 but also some breaking
+changes at the API level. 
+* The `MemoryMeter` constructor and the static methods used to configure the different options (`omitSharedBufferOverhead`, `withGuessing`, `ignoreOuterClassReference`, `ignoreKnownSingletons`, `ignoreNonStrongReferences`, `enableDebug`) have been removed. Instead `MemoryMeter` instances must be created through a `Builder`.
+* The ability to provide a tracker for visited object has been removed.
+* `Guess.NEVER` has been renamed `Guess.ALWAYS_INSTRUMENTATION` for more clarity.
+* `MemoryMeter.countChildren` has been removed.
+* The `MemoryMeter.measure` and `MemoryMeter.measureDeep` now accept `null` parameters
+* The behavior of the `omitSharedBufferOverhead` has been changed as it was incorrect. It was not counting correctly for direct ByteBuffers and considering some buffer has shared even if they were not.
+* Jamm is not trying anymore to support non Hotspot JVM (e.g. OpenJ9)
+* By default `MemoryMeter.measureDeep` is now ignoring the space occupied by known singletons such as `Class` objects, `enums`, `ClassLoaders`, `AccessControlContexts` as well as non-strong references
+(like weak/soft/phantom references). If you want `MemoryMeter` to measure them you need to enable those measurements through `MemoryMeter.builder().measureKnownSingletons()` and `MemoryMeter.builder().measureNonStrongReferences()`.
```

Comment:
> So you opted in for this because you expect most of the time people want to ignore those?
I believe that it is what people will want most of the time they are just not realizing it.

> I am thinking in regards to testing... now we test always ignoring and have two test cases where we explicitly test for those. Wasn't it better to test with not ignoring anything as before and having two cases where we do ignore?
I think that we should test with the default as it is what most users will use.

Reply:
Hmm it seems not in Cassandra though :-)

Reply:
It is actually some bugs in C*. We do not want to measure those.